src/ccm/ucc_ccm.h

```diff
@@ -0,0 +1,103 @@
+* See file LICENSE for terms.
+*/
+
+#ifndef UCC_CCM_H_
+#define UCC_CCM_H_
+
+#include <ucs/config/types.h>
+#include <ucs/debug/log_def.h>
+#include <ucs/config/parser.h>
+#include <assert.h>
+#include <string.h>
+#include "config.h"
+#include "api/ucc.h"
+#include "core/ucc_lib.h"
+#include "utils/ucc_component.h"
+
+typedef struct ucc_ccm_lib     ucc_ccm_lib_t;
+typedef struct ucc_ccm_iface   ucc_ccm_iface_t;
+typedef struct ucc_ccm_context ucc_ccm_context_t;
+typedef struct ucc_ccm_team    ucc_ccm_team_t;
+
+typedef struct ucc_ccm_lib_config {
+    /* Log level above which log messages will be printed */
+    ucs_log_component_config_t log_component;
+    /* Team library priority */
+    int                        priority;
+} ucc_ccm_lib_config_t;
+extern ucs_config_field_t ucc_ccm_lib_config_table[];
+
+typedef struct ucc_ccm_context_config {
+    ucc_ccm_iface_t *iface;
+    ucc_ccm_lib_t   *ccm_lib;
+} ucc_ccm_context_config_t;
+extern ucs_config_field_t ucc_ccm_context_config_table[];
+
+typedef struct ucc_ccm_iface {
+    ucc_component_iface_t          super;
+    char*                          name;
+    int                            priority;
+    ucc_lib_params_t               params;
+    ucs_config_global_list_entry_t ccm_lib_config;
+    ucs_config_global_list_entry_t ccm_context_config;
+    ucc_status_t                   (*init)(const ucc_lib_params_t *params,
+                                           const ucc_lib_config_t *config,
+                                           const ucc_ccm_lib_config_t *ccm_config,
+                                           ucc_ccm_lib_t **ccm_lib);
+    void                           (*cleanup)(ucc_ccm_lib_t *ccm_lib);
+    ucc_status_t                   (*context_create)(ucc_ccm_lib_t *ccm_lib,
+                                                     const ucc_context_params_t *params,
+                                                     const ucc_ccm_context_config_t *config,
+                                                     ucc_ccm_context_t **ccm_context);
+    void                           (*context_destroy)(ucc_ccm_context_t *ccm_context);
+    ucc_status_t                   (*team_create_post)(ucc_ccm_context_t **ccm_ctxs,
+                                                       uint32_t n_ctxs,
+                                                       const ucc_team_params_t *params,
+                                                       ucc_ccm_team_t **team);
+    ucc_status_t                   (*team_create_test)(ucc_ccm_team_t *team_ctx);
+    ucc_status_t                   (*team_destroy)(ucc_ccm_team_t *team);
+} ucc_ccm_iface_t;
+
+typedef struct ucc_ccm_lib {
+    ucc_ccm_iface_t             *iface;
+    ucs_log_component_config_t log_component;
+    int                        priority;
+} ucc_ccm_lib_t;
+
+typedef struct ucc_ccm_context {
+    ucc_ccm_lib_t *ccm_lib;
+} ucc_ccm_context_t;
+
+typedef struct ucc_ccm_team {
+    ucc_ccm_iface_t *iface;
```

Comment:
If "iface" is what I think it is - shouldn't it be part of the API for our "context storage"? I assume this iface is something multiple CCMs/CLs can share?